packages/gatsby/src/utils/webpack.config.js

```diff
@@ -435,32 +449,40 @@ module.exports = async (
           require.resolve(`socket.io-client/package.json`)
         ),
         $virtual: getAbsolutePathForVirtualModule(`$virtual`),
+
+        // Packages used in .cache folder - new resolution can't find them with yarn2/PnP
+        // TODO move to custom resolver
+        mitt: path.dirname(require.resolve(`mitt/package.json`)),
+        "shallow-compare": path.dirname(
+          require.resolve(`shallow-compare/package.json`)
+        ),
+        "gatsby-link": path.dirname(
+          require.resolve(`gatsby-link/package.json`)
+        ),
+        "gatsby-react-router-scroll": path.dirname(
+          require.resolve(`gatsby-react-router-scroll/package.json`)
+        ),
+        "gatsby-legacy-polyfills": path.dirname(
+          require.resolve(`gatsby-legacy-polyfills/package.json`)
+        ),
+        "@mikaelkristiansson/domready": path.dirname(
+          require.resolve(`@mikaelkristiansson/domready/package.json`)
+        ),
+        "@reach/router": path.dirname(
+          require.resolve(`@reach/router/package.json`)
+        ),
+        lodash: path.dirname(require.resolve(`lodash/package.json`)),
+        "prop-types": path.dirname(require.resolve(`prop-types/package.json`)),
```

Comment:
Webpack is not able to find these anymore when using portals in yarn. I should look more into it but we need to test it